src/tasks/receipts.rs

```diff
@@ -0,0 +1,74 @@
+use crate::config::Provider;
+use alloy::{primitives::TxHash, providers::Provider as _};
+use metrics::{counter, histogram};
+use std::time::Instant;
+use tokio::{sync::mpsc, task::JoinHandle};
+use tracing::{debug, error};
+
+/// Submits sidecars in ethereum txns to mainnet ethereum
+#[derive(Debug, Clone)]
+pub struct ReceiptTask {
+    /// Ethereum Provider
+    pub host_provider: Provider,
+}
+
+impl ReceiptTask {
+    pub async fn log_tx(&self, pending_tx_hash: TxHash) {
```

Comment:
nit: let's add a comment to this since it's a public function

Reply:
https://github.com/init4tech/builder/pull/48/commits/7ebd8d219556a9fa66d2c3c697b445c254d6029c